x/ibc/core/02-client/keeper/keeper.go

```diff
@@ -230,6 +230,13 @@ func (k Keeper) ValidateSelfClient(ctx sdk.Context, clientState exported.ClientS
 			tmClient.LatestHeight, ctx.BlockHeight())
 	}
 
+	// consensus params must match consensus params on executing chain
+	expectedConsensusParams := ctx.ConsensusParams()
+	if !reflect.DeepEqual(expectedConsensusParams, tmClient.ConsensusParams) {
```

Comment:
```suggestion
	if expectedConsensusParams == nil || !reflect.DeepEqual(expectedConsensusParams, tmClient.ConsensusParams) {
```

Reply:
I think if the expected consensus params are nil we still want to double check that the set consensus params are also nil. I'm not sure if reflect deep equal does this but I will add a test case and find out

Reply:
see tests added

Reply:
Think colin is correct here, if expected consensus params is nil then provided consensus params should also be nil